pkg/hintrunner/zero/zerohint_memcpy.go

```diff
@@ -28,3 +33,119 @@ func createVMExitScopeHinter(resolver hintReferenceResolver) (hinter.Hinter, err
 		},
 	}, nil
 }
+
+func createSearchSortedLowerHinter(resolver hintReferenceResolver) (hinter.Hinter, error) {
+	array_ptr, err := resolver.GetResOperander("array_ptr")
+	if err != nil {
+		return nil, err
+	}
+	elm_size, err := resolver.GetResOperander("elm_size")
+	if err != nil {
+		return nil, err
+	}
+	n_elms, err := resolver.GetResOperander("n_elms")
+	if err != nil {
+		return nil, err
+	}
+	key, err := resolver.GetResOperander("key")
+	if err != nil {
+		return nil, err
+	}
+	index, err := resolver.GetResOperander("index")
+	if err != nil {
+		return nil, err
+	}
+	return newSearchSortedLowerHint(array_ptr, elm_size, n_elms, key, index), nil
+}
+
+func newSearchSortedLowerHint(array_ptr, elm_size, n_elms, key, index hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "SearchSortedLower",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+
+			//> array_ptr = ids.array_ptr
+			array_ptr, err := hinter.ResolveAsAddress(vm, array_ptr)
+			if err != nil {
+				return err
+			}
+
+			//> elm_size = ids.elm_size
+			elm_size, err := hinter.ResolveAsFelt(vm, elm_size)
+			if err != nil {
+				return err
+			}
+			//> assert isinstance(elm_size, int) and elm_size > 0, \
+			//> 	f'Invalid value for elm_size. Got: {elm_size}.'
+			if elm_size.IsZero() {
+				return fmt.Errorf("Invalid value for elm_size. Got: %v.", elm_size)
+			}
+
+			//> n_elms = ids.n_elms
+			//> assert isinstance(n_elms, int) and n_elms >= 0, \
+			//> 	f'Invalid value for n_elms. Got: {n_elms}.'
+			n_elms, err := hinter.ResolveAsFelt(vm, n_elms)
+
+			if err != nil {
+				return err
+			}
+			if n_elms.IsZero() {
```

Comment:
See here that the condition is `n_elms >= 0`, hence the fact that `n_elms.IsZero()` is true does not contradict the previous fact. I would suggest using `FeltLt` utility instead